modules/host/storageobligations.go

```diff
@@ -131,27 +227,45 @@ func (so *storageObligation) isSane() error {
 		lenMPOs = len(so.RevisionTransactionSet[final].FileContractRevisions[0].NewMissedProofOutputs)
 		if lenVPOs != 2 || lenMPOs != 2 {
 			build.Critical("file contract has wrong number of VPOs and MPOs, expecting 2 each:", lenVPOs, lenMPOs)
-			return ErrInsaneFileContractRevisionOutputCounts
+			return errInsaneFileContractRevisionOutputCounts
 		}
 	}
 	return nil
 }
 
-// id returns the id of the storage obligation, which is definied by the file
-// contract id of the file contract that governs the storage contract.
-func (so *storageObligation) id() types.FileContractID {
-	err := so.isSane()
-	if err != nil {
-		build.Critical("insane storage obligation when returning id")
-		return types.FileContractID{}
+// payous returns the set of valid payouts and missed payouts that represent
+// the latest revision for the storage obligation.
+func (so *storageObligation) payouts() (valid []types.SiacoinOutput, missed []types.SiacoinOutput) {
+	valid = make([]types.SiacoinOutput, 2)
+	missed = make([]types.SiacoinOutput, 2)
+	if len(so.RevisionTransactionSet) > 0 {
+		copy(valid, so.RevisionTransactionSet[len(so.RevisionTransactionSet)-1].FileContractRevisions[0].NewValidProofOutputs)
+		copy(missed, so.RevisionTransactionSet[len(so.RevisionTransactionSet)-1].FileContractRevisions[0].NewMissedProofOutputs)
+		return
 	}
-	return so.OriginTransactionSet[len(so.OriginTransactionSet)-1].FileContractID(0)
+	copy(valid, so.OriginTransactionSet[len(so.OriginTransactionSet)-1].FileContracts[0].ValidProofOutputs)
+	copy(missed, so.OriginTransactionSet[len(so.OriginTransactionSet)-1].FileContracts[0].MissedProofOutputs)
+	return
+}
+
+// proofDeadline returns the height by which the storage proof must be
+// submitted.
+func (so *storageObligation) proofDeadline() types.BlockHeight {
+	if len(so.RevisionTransactionSet) > 0 {
+		return so.RevisionTransactionSet[len(so.RevisionTransactionSet)-1].FileContractRevisions[0].NewWindowEnd
+	}
+	return so.OriginTransactionSet[len(so.OriginTransactionSet)-1].FileContracts[0].WindowEnd
```

Comment:
this sort of logic crops up in the contractor as well. It might be worth defining a single type for handling contract+revision. Something like:
```go
type RevisedContract struct {
    OriginalContract types.FileContract
    LastRevision   types.FileContractRevision
}
```
I'm not sure what its methods would look like, but maybe:
```go
func (rc RevisedContract) AfterRevision() types.FileContract {
    // If there is a revision, apply it to the original contract.
    // Otherwise just return the original.
}
```